client/src/components/molecules/ClientRequestTable.tsx

```diff
@@ -0,0 +1,229 @@
+  useEffect(() => {
+    const requests = props.requests.filter((request) => !request.deleted);
+    const nonFulfilledRequests = requests.filter((request) => {
+      if (request !== undefined) {
+        if (request.fulfilled === false) {
+          return request;
+        }
+      }
+    });
+    const fulfilledRequests = requests.filter((request) => {
+      if (request !== undefined) {
+        if (request.fulfilled === true) {
+          return request;
+        }
+      }
+    });
+
+    nonFulfilledRequests.sort((a, b) => {
+      return a!.dateCreated!.valueOf() - b!.dateCreated!.valueOf();
+    });
+
+    fulfilledRequests.sort((a, b) => {
+      return a!.dateCreated!.valueOf() - b!.dateCreated!.valueOf();
+    });
+
+    const sortedRequests: Request[] = nonFulfilledRequests!.concat(
+      fulfilledRequests!
+    ) as Request[];
+    setRequests(sortedRequests);
+  }, [props.requests]);
+
+  const [mutateDeleteRequest] = useMutation(softDeleteRequest);
+  const [mutateRequest] = useMutation(updateRequest);
+
+  const onSoftDeleteRequest = (index: number) => {
+    const requestsCopy = requests.slice();
+    const req = { ...requestsCopy[index] };
+    requestsCopy.splice(index, 1);
+    const id = req._id;
+    setRequests(requestsCopy);
+    props.onChangeRequests!(requestsCopy);
+    mutateDeleteRequest({ variables: { id: id } });
+  };
+
+  const onFulfilledRequest = (index: number) => {
+    const requestsCopy = requests.slice();
+    const req = { ...requestsCopy[index] };
+    if (req.fulfilled === false) {
+      req.fulfilled = true;
+      requestsCopy.splice(index, 1);
+      let i = requestsCopy.length - 1;
+      for (; i > -1; --i) {
+        if (requestsCopy[i].fulfilled === false) break;
+        else if (
+          requestsCopy[i]!.dateCreated!.valueOf() < req!.dateCreated!.valueOf()
+        )
+          break;
+      }
+      requestsCopy.splice(i + 1, 0, req);
+    } else {
+      req.fulfilled = false;
+      requestsCopy.splice(index, 1);
+      let i = 0;
+      for (; i < requestsCopy.length; ++i) {
+        if (requestsCopy[i].fulfilled === true) break;
+        else if (
+          requestsCopy[i]!.dateCreated!.valueOf() > req!.dateCreated!.valueOf()
+        )
+          break;
+      }
+      requestsCopy.splice(i, 0, req);
+    }
+    setRequests(requestsCopy);
+    props.onChangeRequests!(requestsCopy);
+    const id = req._id;
+    const requestId = req.requestId;
+    const fulfilled = req.fulfilled;
+    mutateRequest({ variables: { request: { id, requestId, fulfilled } } });
+  };
+
+  return (
+    <div className="request-list">
+      {requestSelectedForEditing && (
+        <RequestForm
+          onSubmitComplete={() => {
+            window.location.reload();
+          }}
+          handleClose={() => setRequestSelectedForEditing("")}
+          operation="edit"
+          requestId={requestSelectedForEditing}
+        />
+      )}
+      {requests.length > 0 && (
```

Comment:
nit: 

I think the table component should be able to render regardless of the amount of requests. It should be up to the parent component whether or not it displays a table with no requests. 

Removing the conditional rendering here would improve the usability of the request table component

this is just a fyi :))))) since we're not actually going to be building more pages using this component

Reply:
oh that's a good point! i'll keep that in mind